### PR TITLE
feat(jans-lock): structuring Cedarling MVP according to the anatomy provided

### LINKED ISSUE #9353: feat (cedarling): structuring Cedarling MVP according to the anatomy provided.
- [ ] Cedarling should communicate with the external world only using two interfaces init and authz.
- [ ] It will be great to have packages based on the engine's naming conventions (containing the code as per the engine's responsibility). It will be especially helpful to navigate code as the application grows 

image

### PR DESCRIPTION
**Pull request by olehbozhok**
### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

### Description
Structuring Cedarling MVP according to the anatomy provided.

#### Target issue
[link](https://github.com/JanssenProject/jans/issues/9353)
  
closes #9353 

#### Implementation Details
  Created folder `engines`. Added to it: 
- authz_engine
- init_engine
- jwt_engine
other engines currently are not exist in current codebase.

Created folder `bindings`. And moved folder `cedarling_python`.

Added unit tests.  
Certain test scenarios were intentionally ignored because there are cases where an error propagates due to the API, but in reality, the error won’t occur in 99% of cases. So, it’s very difficult to reproduce.

**Code line coverage** is 80%+ for files where we have business logic.  
File `engines\jwt_engine\src\lib.rs` was skipped during test writing. Because it has no correct implementation. What will be changed in the future.

The code coverage was checked using `cargo-llvm-cov`  
Use next command to install
```
cargo install cargo-llvm-cov  
```
and next command to watch the results:
```
cargo llvm-cov --html --open
```

Information about `cargo-llvm-cov` was added to README file.

-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated
- [x] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [ ] **I confirm that there is no impact on the docs due to the code changes in this PR.**

### CODE REVIEW

File: jans-lock/cedarling/Cargo.toml [hunk 1]
```diff
@@ -1,11 +1,19 @@
 [workspace]
 resolver = "2"
-members = ["authz", "authz_run", "cedarling_python", "init_engine", "jwt"]
```
Comment (SafinWasi) [inline jans-lock/cedarling/Cargo.toml:3]: In the different parts of the documentation, the old crate name `authz_run` is still being used.

File: jans-lock/cedarling/engines/init_engine/src/lib.rs [hunk 2]
```diff
@@ -26,7 +26,7 @@ pub struct BootstrapConfig {
 }
 
 impl BootstrapConfig {
-	pub fn get_jwt_decoder(&self) -> jwt::JWTDecoder {
-		jwt::JWTDecoder::WithoutValidation
+	pub fn get_jwt_decoder(&self) -> jwt_engine::JWTDecoder {
+		jwt_engine::JWTDecoder::WithoutValidation
```
Comment (SafinWasi) [inline jans-lock/cedarling/engines/init_engine/src/lib.rs:30]: Are we not validating JWTs? I looked at crate `jwt_engine` and it does not have implementation for validating JWT.

**Review by SafinWasi — APPROVED** [review]
Code looks good to me, but I have a few questions, please check.
Also, since we have multiple library crates, I would like a readme for each crate with a brief description of functionality, and a list of library crates in the main readme.

File: jans-lock/cedarling/engines/init_engine/src/lib.rs (hunk 2 above)
Comment (olehbozhok) [reply jans-lock/cedarling/engines/init_engine/src/lib.rs:30]: Yes, It is. Validation of JWT will be implemented in the future.

File: jans-lock/cedarling/Cargo.toml (hunk 1 above)
Comment (olehbozhok) [reply jans-lock/cedarling/Cargo.toml:3]: Thanks! Pushed the commit that fix it.

**Review by SafinWasi — DISMISSED** [review]

**Review by duttarnab — DISMISSED** [review]

**Review by moabu — CHANGES_REQUESTED** [review]
Please specify the edition and versions correctly. Due to the monorepo policy the edition must start from the version it starts with in janssen so in this case it would be `1.1.5`

**Review by moabu — APPROVED** [review]

**Pull request merged**